lib/core/src/sync/mod.rs

```diff
@@ -208,6 +227,124 @@ impl SyncService {
 
         Ok(())
     }
+
+    async fn push(&self) -> Result<()> {
+        let outgoing_details = self.persister.get_sync_outgoing_details()?;
+
+        for SyncOutgoingDetails {
+            record_id,
+            record_type,
+            ..
+        } in outgoing_details
+        {
+            // Step 1: Get the sync state, ensure it exists
+            let Some(sync_state) = unwrap_or_continue!(
```

Comment:
Since we are iterating each record here I don't think we need the unwrap_or_continue macro repetitive calls.
We can have one method that is called process_push for example which we call here and log on error once if we get error as result.
I think it sill simplify the code.